aiida_firecrest/transport.py

```diff
@@ -363,77 +642,227 @@ def getfile(
+            raise ValueError(
+                f"Checksum mismatch between local and remote files: {local} and {remote}"
+            ) from e
+
+    def _gettreetar(
+        self,
+        remotepath: str | FcPath,
+        localpath: str | Path,
+        dereference: bool = False,
+        *args: Any,
+        **kwargs: Any,
+    ) -> None:
+        """Get a directory from the remote as a tar file and extract it locally.
+        This is useful for downloading a directory with many files,
+        as it might be more efficient than downloading each file individually.
+        Note that this method is not part of the Transport interface, and is not meant to be used publicly.
+
+        :param dereference: If True, follow symlinks.
+            note: FirecREST doesn't support `--dereference` for tar call,
+            so dereference should always be False, for now.
+        """
+
+        _ = uuid.uuid4()
+        remote_path_temp = self._temp_directory.joinpath(f"temp_{_}.tar")
+
+        # Compress
+        self._client.compress(self._machine, str(remotepath), remote_path_temp)
+
+        # Download
+        localpath_temp = Path(localpath).joinpath(f"temp_{_}.tar")
+        try:
+            self.getfile(remote_path_temp, localpath_temp)
+        finally:
+            self.remove(remote_path_temp)
+
+        # Extract the downloaded file locally
+        try:
+            # with tarfile.open(localpath_temp, "r") as tar:
+            #     members = [m for m in tar.getmembers() if m.name.startswith(remotepath.name)]
+            #     for member in members:
+            #         member.name = os.path.relpath(member.name, remotepath.name)
+            #         tar.extract(member, path=localpath)
+            os.system(f"tar -xf '{localpath_temp}' --strip-components=1 -C {localpath}")
+        finally:
+            localpath_temp.unlink()
 
     def gettree(
-        self, remotepath: str | FcPath, localpath: str | Path, *args: Any, **kwargs: Any
+        self,
+        remotepath: str | FcPath,
+        localpath: str | Path,
+        dereference: bool = True,
+        *args: Any,
+        **kwargs: Any,
     ) -> None:
+        """Get a directory from the remote.
+
+        :param dereference: If True, follow symlinks.
+            note: dereference should be always True, otherwise the symlinks will not be functional.
+        """
+
         local = Path(localpath)
         if not local.is_absolute():
             raise ValueError("Destination must be an absolute path")
         if local.is_file():
             raise OSError("Cannot copy a directory into a file")
-        local.mkdir(parents=True, exist_ok=True)
 
         remote = (
             remotepath
             if isinstance(remotepath, FcPath)
-            else self._cwd.joinpath(remotepath).enable_cache()
+            else self._cwd.joinpath(
+                remotepath
+            )  # .enable_cache() it's removed from from path.py to be investigated
```

Comment:
unclear, please consider to make an issue

Reply:
Alright, just opened an issue is opened on this:
https://github.com/aiidateam/aiida-firecrest/issues/41